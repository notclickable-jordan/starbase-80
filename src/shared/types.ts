export interface IServiceCategory {
	category: string;
	bubble?: boolean;
	bubbleBGLight?: string;
	bubbleBGDark?: string;
	services: IService[];
	iconBubblePadding?: boolean;
}

export interface IService {
	name: string;
	uri: string;

	description?: string;
	icon?: string;
	iconLight?: string;
	iconDark?: string;
	iconColor?: string;
	iconBG?: string;
	iconBubble?: boolean;
	iconBubblePadding?: boolean;
	iconAspect?: IconAspect;

	newWindow?: boolean;
}

export type IconAspect = "square" | "width" | "height";
