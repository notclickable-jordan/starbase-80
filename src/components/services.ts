import { is } from "../shared/is";
import { IService } from "../shared/types";
import { Anchor } from "./anchor";
import { Icon } from "./icon";

interface IServicesProps {
	services: IService[];
	categoryBubblePadding?: boolean;
}

export const Services = function (props: IServicesProps) {
	const { services, categoryBubblePadding } = props;

	return `
		<ul class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-1 lg:gap-2 lg:gap-y-4">
			${services.map((service, index) => Service({ index, service, categoryBubblePadding })).join("")}
		</ul>
	`;
};

interface IServiceProps {
	service: IService;
	index: number;
	categoryBubblePadding?: boolean;
}

function Service(props: IServiceProps) {
	const { service, index, categoryBubblePadding } = props;

	const {
		name,
		uri,
		description,
		icon,
		iconLight,
		iconDark,
		iconBG,
		iconBubble,
		iconBubblePadding,
		iconColor,
		iconAspect,
		newWindow,
	} = service;

	return `
		<li class="p-4">
		${Anchor({
			uri,
			newWindow,
			className: "flex gap-4",
			children: `${
				!is.null(icon)
					? `<span class="flex-shrink-0 flex">${Icon({
							name,
							icon,
							iconLight,
							iconDark,
							uri,
							index,
							iconColor,
							iconBG,
							iconBubble,
							iconAspect,
							newWindow,
							categoryBubblePadding,
							iconBubblePadding,
						})}</span>`
					: ``
			}
		<div>
			<h3 class="text-lg mt-1 font-semibold line-clamp-1">${name}</h3>
			${!is.null(description) ? `<p class="text-sm text-black/50 dark:text-white/50 line-clamp-1">${description}</p>` : ``}
		</div>`,
		})}
			
		</li>
	`;
}
