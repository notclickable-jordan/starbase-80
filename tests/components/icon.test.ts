import { Icon } from "../../src/components/icon";
import { IconAspect } from "../../src/shared/types";

describe("Icon Component", () => {
	const testColors = ["blue", "rose", "green", "red", "yellow"];
	const longString =
		"This is a very long service name that tests how the component handles extensive text content and ensures that it properly renders without breaking the layout or functionality";

	describe("Required props only", () => {
		test("should render with just name and index", () => {
			const result = Icon({
				name: "Test Service",
				index: 0,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with long name", () => {
			const result = Icon({
				name: longString,
				index: 1,
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With icon variations", () => {
		test("should render with HTTP URL icon", () => {
			const result = Icon({
				name: "External Service",
				index: 0,
				icon: "https://example.com/icon.png",
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with local path icon", () => {
			const result = Icon({
				name: "Local Service",
				index: 1,
				icon: "/icons/service.png",
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with material design icon", () => {
			const result = Icon({
				name: "Material Service",
				index: 2,
				icon: "mdi-home",
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with dashboard icon", () => {
			const result = Icon({
				name: "Dashboard Service",
				index: 3,
				icon: "nextcloud",
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with selfhst icon", () => {
			const result = Icon({
				name: "SelfHst Service",
				index: 4,
				icon: "selfhst-jellyfin",
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With iconColor variations", () => {
		testColors.forEach((color, index) => {
			test(`should render with ${color} iconColor`, () => {
				const result = Icon({
					name: `${color} Service`,
					index: index,
					icon: "mdi-star",
					iconColor: color,
				});
				expect(result).toMatchSnapshot();
			});
		});

		test("should render with hex color", () => {
			const result = Icon({
				name: "Hex Color Service",
				index: 0,
				icon: "mdi-palette",
				iconColor: "#FF6B35",
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With iconBG variations", () => {
		testColors.forEach((color, index) => {
			test(`should render with ${color} iconBG`, () => {
				const result = Icon({
					name: `${color} BG Service`,
					index: index,
					icon: "/icon.png",
					iconBG: `${color}-100`,
				});
				expect(result).toMatchSnapshot();
			});
		});

		test("should render with hex background", () => {
			const result = Icon({
				name: "Hex BG Service",
				index: 0,
				icon: "/icon.png",
				iconBG: "#E8F4FD",
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With iconBubble variations", () => {
		test("should render with iconBubble true", () => {
			const result = Icon({
				name: "Bubble Service",
				index: 0,
				icon: "/icon.png",
				iconBubble: true,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with iconBubble false", () => {
			const result = Icon({
				name: "No Bubble Service",
				index: 1,
				icon: "/icon.png",
				iconBubble: false,
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With iconBubblePadding variations", () => {
		test("should render with iconBubblePadding true", () => {
			const result = Icon({
				name: "Padded Service",
				index: 0,
				icon: "/icon.png",
				iconBubblePadding: true,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with iconBubblePadding false", () => {
			const result = Icon({
				name: "Unpadded Service",
				index: 1,
				icon: "/icon.png",
				iconBubblePadding: false,
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With iconAspect variations", () => {
		const aspects: IconAspect[] = ["square", "width", "height"];

		aspects.forEach((aspect, index) => {
			test(`should render with iconAspect ${aspect}`, () => {
				const result = Icon({
					name: `${aspect} Service`,
					index: index,
					icon: "/icon.png",
					iconAspect: aspect,
				});
				expect(result).toMatchSnapshot();
			});
		});
	});

	describe("With uri and newWindow variations", () => {
		test("should render with uri", () => {
			const result = Icon({
				name: "Linked Service",
				index: 0,
				icon: "/icon.png",
				uri: "https://example.com",
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with uri and newWindow true", () => {
			const result = Icon({
				name: "New Window Service",
				index: 0,
				icon: "/icon.png",
				uri: "https://example.com",
				newWindow: true,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with uri and newWindow false", () => {
			const result = Icon({
				name: "Same Window Service",
				index: 0,
				icon: "/icon.png",
				uri: "https://example.com",
				newWindow: false,
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("With categoryBubblePadding variations", () => {
		test("should render with categoryBubblePadding true", () => {
			const result = Icon({
				name: "Category Padded Service",
				index: 0,
				icon: "/icon.png",
				categoryBubblePadding: true,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with categoryBubblePadding false", () => {
			const result = Icon({
				name: "Category Unpadded Service",
				index: 0,
				icon: "/icon.png",
				categoryBubblePadding: false,
			});
			expect(result).toMatchSnapshot();
		});
	});

	describe("Combined prop variations", () => {
		test("should render with some optional props", () => {
			const result = Icon({
				name: "Partial Service",
				index: 0,
				icon: "mdi-cog",
				iconColor: "blue",
				iconBubble: true,
			});
			expect(result).toMatchSnapshot();
		});

		testColors.forEach((color, index) => {
			test(`should render with ${color} theme combination`, () => {
				const result = Icon({
					name: `${color.charAt(0).toUpperCase() + color.slice(1)} Service`,
					index: index,
					icon: "mdi-star",
					iconColor: color,
					iconBG: `${color}-100`,
					iconBubble: true,
					iconBubblePadding: index % 2 === 0,
					iconAspect: index % 3 === 0 ? "square" : index % 3 === 1 ? "width" : "height",
					uri: `https://${color}.example.com`,
					newWindow: index % 2 === 0,
				});
				expect(result).toMatchSnapshot();
			});
		});

		test("should render with all props", () => {
			const result = Icon({
				name: "Complete Service",
				index: 0,
				icon: "https://example.com/complete-icon.png",
				iconColor: "indigo",
				iconBG: "indigo-50",
				iconBubble: true,
				iconBubblePadding: true,
				iconAspect: "square",
				uri: "https://complete.example.com",
				newWindow: true,
				categoryBubblePadding: true,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render with all props using long strings", () => {
			const result = Icon({
				name: longString,
				index: 99,
				icon: "https://very-long-domain-name.example.com/path/to/very/long/icon/filename/with/lots/of/descriptive/text.png",
				iconColor: "#FF6B35",
				iconBG: "#E8F4FD",
				iconBubble: true,
				iconBubblePadding: true,
				iconAspect: "width",
				uri: "https://very-long-domain-name-for-testing.example.com/with/a/very/long/path",
				newWindow: false,
				categoryBubblePadding: false,
			});
			expect(result).toMatchSnapshot();
		});

		test("should render without icon (blank icon)", () => {
			const result = Icon({
				name: "Blank Icon Service",
				index: 5,
				uri: "https://example.com",
			});
			expect(result).toMatchSnapshot();
		});

		test("should render without icon and without uri", () => {
			const result = Icon({
				name: "Static Blank Service",
				index: 7,
			});
			expect(result).toMatchSnapshot();
		});
	});
});
