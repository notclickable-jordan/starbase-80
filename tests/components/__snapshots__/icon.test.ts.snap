// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Icon Component Combined prop variations should render with all props 1`] = `
"
			<a href="https://complete.example.com" target="_blank" rel="noopener noreferrer" title="Complete Service" class="">
				<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-indigo-50" style=""><img src="https://example.com/complete-icon.png" alt="" class="w-14 h-14 object-contain" /></span>
			</a>
		"
`;

exports[`Icon Component Combined prop variations should render with all props using long strings 1`] = `
"
		<a href="https://very-long-domain-name-for-testing.example.com/with/a/very/long/path" title="This is a very long service name that tests how the component handles extensive text content and ensures that it properly renders without breaking the layout or functionality" class="">
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 rounded-2xl border border-black/5 shadow-sm" style="background-color: #E8F4FD"><img src="https://very-long-domain-name.example.com/path/to/very/long/icon/filename/with/lots/of/descriptive/text.png" alt="" class="w-14" /></span>
		</a>
	"
`;

exports[`Icon Component Combined prop variations should render with blue theme combination 1`] = `
"
			<a href="https://blue.example.com" target="_blank" rel="noopener noreferrer" title="Blue Service" class="">
				
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-blue-100" style="">
				<span>
					<span
						class="flex items-center justify-center w-14 h-14 object-contain bg-blue overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			
			</a>
		"
`;

exports[`Icon Component Combined prop variations should render with green theme combination 1`] = `
"
			<a href="https://green.example.com" target="_blank" rel="noopener noreferrer" title="Green Service" class="">
				
			<span class="flex items-center justify-center overflow-hidden bg-contain h-16 rounded-2xl border border-black/5 shadow-sm bg-green-100" style="">
				<span>
					<span
						class="flex items-center justify-center h-14 bg-green overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			
			</a>
		"
`;

exports[`Icon Component Combined prop variations should render with red theme combination 1`] = `
"
		<a href="https://red.example.com" title="Red Service" class="">
			
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-red-100" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-red overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			
		</a>
	"
`;

exports[`Icon Component Combined prop variations should render with rose theme combination 1`] = `
"
		<a href="https://rose.example.com" title="Rose Service" class="">
			
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 rounded-2xl border border-black/5 shadow-sm bg-rose-100" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 bg-rose overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			
		</a>
	"
`;

exports[`Icon Component Combined prop variations should render with some optional props 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-blue overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/cog.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/cog.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component Combined prop variations should render with yellow theme combination 1`] = `
"
			<a href="https://yellow.example.com" target="_blank" rel="noopener noreferrer" title="Yellow Service" class="">
				
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 rounded-2xl border border-black/5 shadow-sm bg-yellow-100" style="">
				<span>
					<span
						class="flex items-center justify-center w-14 bg-yellow overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			
			</a>
		"
`;

exports[`Icon Component Combined prop variations should render without icon (blank icon) 1`] = `
"
			<a href="https://example.com" target="_blank" rel="noopener noreferrer" title="Blank Icon Service" class="">
				
		<span
			class="block w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-blue-300 overflow-hidden"
		/>
	
			</a>
		"
`;

exports[`Icon Component Combined prop variations should render without icon and without uri 1`] = `
"
		<span
			class="block w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-rose-300 overflow-hidden"
		/>
	"
`;

exports[`Icon Component Required props only should render with just name and index 1`] = `
"
		<span
			class="block w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-undefined-300 overflow-hidden"
		/>
	"
`;

exports[`Icon Component Required props only should render with long name 1`] = `
"
		<span
			class="block w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-blue-300 overflow-hidden"
		/>
	"
`;

exports[`Icon Component With categoryBubblePadding variations should render with categoryBubblePadding false 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With categoryBubblePadding variations should render with categoryBubblePadding true 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-14 h-14 object-contain" /></span>"`;

exports[`Icon Component With icon variations should render with HTTP URL icon 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="https://example.com/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With icon variations should render with dashboard icon 1`] = `
"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style="">
				<img
					src=https://cdn.jsdelivr.net/gh/walkxcode/dashboard-icons/png/nextcloud.png
					alt=""
					class="w-16 h-16 object-contain"
				/>
			</span>"
`;

exports[`Icon Component With icon variations should render with local path icon 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icons/service.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With icon variations should render with material design icon 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-black dark:bg-white overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/home.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/home.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With icon variations should render with selfhst icon 1`] = `
"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style="">
				<img
					src=https://cdn.jsdelivr.net/gh/selfhst/icons/png/jellyfin.png
					alt=""
					class="w-16 h-16 object-contain"
				/>
			</span>"
`;

exports[`Icon Component With iconAspect variations should render with iconAspect height 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="h-16" /></span>"`;

exports[`Icon Component With iconAspect variations should render with iconAspect square 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconAspect variations should render with iconAspect width 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16" /></span>"`;

exports[`Icon Component With iconBG variations should render with blue iconBG 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-blue-100" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBG variations should render with green iconBG 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-green-100" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBG variations should render with hex background 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style="background-color: #E8F4FD"><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBG variations should render with red iconBG 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-red-100" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBG variations should render with rose iconBG 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-rose-100" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBG variations should render with yellow iconBG 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-yellow-100" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBubble variations should render with iconBubble false 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBubble variations should render with iconBubble true 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBubblePadding variations should render with iconBubblePadding false 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>"`;

exports[`Icon Component With iconBubblePadding variations should render with iconBubblePadding true 1`] = `"<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-14 h-14 object-contain" /></span>"`;

exports[`Icon Component With iconColor variations should render with blue iconColor 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-blue overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With iconColor variations should render with green iconColor 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-green overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With iconColor variations should render with hex color 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain  overflow-hidden"
						style="background-color: #FF6B35;mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/palette.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/palette.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With iconColor variations should render with red iconColor 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-red overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With iconColor variations should render with rose iconColor 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-rose overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With iconColor variations should render with yellow iconColor 1`] = `
"
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm bg-slate-200 dark:bg-gray-900" style="">
				<span>
					<span
						class="flex items-center justify-center w-16 h-16 object-contain bg-yellow overflow-hidden"
						style="mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain;webkit-mask: url(https://cdn.jsdelivr.net/npm/@mdi/svg@latest/svg/star.svg) no-repeat center / contain"
					></span>
				</span>
				</span>
			"
`;

exports[`Icon Component With uri and newWindow variations should render with uri 1`] = `
"
			<a href="https://example.com" target="_blank" rel="noopener noreferrer" title="Linked Service" class="">
				<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>
			</a>
		"
`;

exports[`Icon Component With uri and newWindow variations should render with uri and newWindow false 1`] = `
"
		<a href="https://example.com" title="Same Window Service" class="">
			<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>
		</a>
	"
`;

exports[`Icon Component With uri and newWindow variations should render with uri and newWindow true 1`] = `
"
			<a href="https://example.com" target="_blank" rel="noopener noreferrer" title="New Window Service" class="">
				<span class="flex items-center justify-center overflow-hidden bg-contain w-16 h-16 rounded-2xl border border-black/5 shadow-sm" style=""><img src="/icon.png" alt="" class="w-16 h-16 object-contain" /></span>
			</a>
		"
`;
